buildAddOn.py

```diff
@@ -0,0 +1,253 @@
+    configData = json.load(configFile)
+    addOnName = configData['addOnName']
+
+    optionalParams = None
+    if 'addOnSpecificCMakeParameters' in configData:
+        optionalParams = configData['addOnSpecificCMakeParameters']
+
+    if args.language:
+        configUpper = [lang.upper() for lang in configData['languages']]
+        languageList = [lang.upper() for lang in args.language]
+        if 'ALL' in languageList:
+            languageList = configUpper
+        else:
+            for lang in languageList:
+                if lang not in configUpper:
+                    print('Language not supported')
+                    return 1
+
+    # Create directory for Build and Package
+    rootFolder = pathlib.Path(__file__).parent.parent.absolute()
+    buildFolder = rootFolder / 'Build'
+    packageRootFolder = buildFolder / 'Package' / addOnName
+    devKitFolderList = {}
+
+    os.chdir (rootFolder)
+
+    if PrepareDirectories(rootFolder, buildFolder, packageRootFolder, devKitFolderList, args, configData, platformName) != 0:
+        return 1
+
+    # At this point, devKitFolderList dictionary has all provided ACVersions as keys
+    # For every ACVersion
+    # If release, build Add-On for all languages with RelWithDebInfo configuration
+    # Else build Add-On with Debug and RelWithDebInfo configurations, without language specified   
+    # In each case, if package creation is enabled, copy the .apx/.bundle files to the Package directory
+    for version in devKitFolderList:
+        devKitFolder = devKitFolderList[version]
+
+        if args.release is True:
+            for languageCode in languageList:
+                if BuildAddOn(rootFolder, buildFolder, devKitFolder, version, configData, platformName, 'RelWithDebInfo', languageCode) != 0:
+                    return 1
+                if args.package:
+                    CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, 'RelWithDebInfo', languageCode, True)
+
+        else:
+            if BuildAddOn(rootFolder, buildFolder, devKitFolder, version, configData, platformName, 'Debug') != 0:
+                return 1
+            if BuildAddOn(rootFolder, buildFolder, devKitFolder, version, configData, platformName, 'RelWithDebInfo') != 0:
+                return 1
+            if args.package:
+                CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, 'Debug')
+                CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, 'RelWithDebInfo')
+
+    # Zip packages
+    if args.package:
+        release = 'Release_Localized' if args.release else 'Daily'
+        for version in args.acVersion:
+            subprocess.call ([
+                '7z', 'a',
+                str(packageRootFolder.parent / f'{addOnName}-{version}_{release}_{platformName}.zip'),
+                str(packageRootFolder / version / '*')
+            ])
+
+    print ('Build Successful')

```

Comment:
Lets use 'Build succeeded!' to keep the messages consistent with Archicad build messages. (and lets use the exclamation mark at the end of error messages as well in this spirit)

Reply:
Fixed